### PR TITLE
Split World Location News out of World Locations

### PR DESCRIPTION
**Pull request by brucebolt**
The `WorldLocation` model currently contains the data for two published pages: the World Location itself and the World Location News page.

This has led to some unusual code needing to be written specifically for the news page as publishing the `WorldLocation` model using `PublishesToPublishingApi` and `Searchable` will lead to only the World Location page being published.

By splitting the model into two parts, we will be able to remove this custom code and re-use existing methods for publishing the news page.

[Trello card](https://trello.com/b/NHxZVLsg)

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

### CODE REVIEW

**Comment by BeckaL** [comment]
For this todo (you won't be able to see it in this code once I remove the commit that adds the TODO)

TODO 1/3:

```
# TODO: what do we actually need to send to search for world location
  searchable title: :title,
             link: :search_link,
             description: :search_description,
             only: :active_international_delegation,
             format: "world_location",
             slug: :slug
```

I think I understand things very slightly more having looked into international delegations and their news pages. We only really publish world locations that are of the type 'international delegation', hence the code in this stuff we send to search. I think we should do a similiar refactoring to what we've done here for international delegations, splitting them out of world locations (and dropping their news pages which I don't think are useful - pending a product decisions). So I'm voting to keep this in for now, and address as part of addressing international delegations. Anybody feel free to shout if they think we _should_ do something as part of this.

**Comment by BeckaL** [comment]
TODOs 2 and 3 / 3:

```
  # TODO: what do we actually need to send to publishing api for world location
  include PublishesToPublishingApi

  # TODO: why does this override the method in PublishesToPublishingApi
  def publish_to_publishing_api
    # WorldLocations no longer support translations
    # but the current world news pages use the world location object
    # to build their featured articles lists so we need to keep
    # the translations but not publish them.
    run_callbacks :published do
      PublishingApiWorker.perform_async(
        self.class.name,
        id,
        nil,
        "en",
      )
    end
  end
```
I'm also proposing we leave these as they are and address as part of future work to rationalise world locations / international delegations.

I basically think we should be able to split out international delegations and remove world locations as a model entirely. But not addressing them as part of this.

**Comment by BeckaL** [comment]
Have just broken the tests again, I think there's one last bit of search behaviour that's off in that we're sending translated news pages to search now, will look at this tomorrow. But otherwise think this _should_ be a passing build

File: app/models/searchable.rb
Comment (brucebolt) [inline app/models/searchable.rb:151]: I wonder how this works for organisations, as these have translations but aren't included in search?

Comment (brucebolt) [reply app/models/searchable.rb]: I've come up with a solution that doesn't seem to affect any of the other document types.  I'm not sure why we're not filtering for other documents 😕 .

File: test/integration/world_location_integration_test.rb
Comment (JonathanHallam) [inline test/integration/world_location_integration_test.rb:82]: totally not the place for this chat, but what's an inactive location?

Comment (brucebolt) [reply test/integration/world_location_integration_test.rb:82]: From what I can see, we've imported the list of World Locations from somewhere.  For some of these (e.g. the UK), we don't want a world location page, so they are marked as 'inactive'.  Although we weirdly still have a location page published, presumably for legacy reasons 😕.

File: app/controllers/admin/world_location_news_controller.rb
Comment (JonathanHallam) [inline app/controllers/admin/world_location_news_controller.rb:54]: is this meant to still be here?

File: app/models/world_location.rb
Comment (JonathanHallam) [inline app/models/world_location.rb:17]: Really like how this cuts down the complexity of the model. I'm assuming the featured/offsite links were only used in the previously coupled world location news bit of this

File: app/presenters/publishing_api/world_location_news_presenter.rb
Comment (JonathanHallam) [inline app/presenters/publishing_api/world_location_news_presenter.rb:7]: suuuuper minor and I know it was done this way already, but what's the advantage of passing in `nil` here rather than `"major"`, would mean line 10 could be `self.update_type = update_type`

File: app/views/admin/world_location_news/edit.html.erb
Comment (JonathanHallam) [inline app/views/admin/world_location_news/edit.html.erb:6]: Should this be "changes to world location news articles appear instantly...." or do we not care about publishers knowing the difference?

File: app/controllers/admin/world_location_news_controller.rb
Comment (brucebolt) [reply app/controllers/admin/world_location_news_controller.rb:54]: Nope.  I'll remove and rebase.

File: app/models/world_location.rb
Comment (brucebolt) [reply app/models/world_location.rb:17]: Yes, they were only used for the news part.  The rest of the `WorldLocation` didn't use them at all.

File: app/views/admin/world_location_news/edit.html.erb
Comment (brucebolt) [reply app/views/admin/world_location_news/edit.html.erb:6]: We've kept [the previous wording](https://github.com/alphagov/whitehall/blob/26f54ba8ac13ef5c2da8ba4c14a6dd1b610ded0d/app/views/admin/world_locations/edit.html.erb#L6) as the content being published is exactly the same as before.

The whole interface around world locations probably needs revisiting from a design perspective, so maybe we could ask Publishing Experience to put this on their backlog?

File: test/integration/world_location_integration_test.rb
Comment (brucebolt) [reply test/integration/world_location_integration_test.rb]: Hopefully we can look at [this Trello card](https://trello.com/c/rC4EjFOV/125-rename-references-to-rummager-in-whitehall-to-search-api) sometime soon, so all the references can be finally removed.

File: app/presenters/publishing_api/world_location_news_presenter.rb
Comment (brucebolt) [reply app/presenters/publishing_api/world_location_news_presenter.rb:7]: We looked into this offline and found it isn't possible.  The `update_type` will always be `major`, as this is non-editionable content, so we think further work should be done (separately) to refactor the non-editionable presenters so this value isn't needed to be defined at all.

File: config/routes.rb
Comment (JonathanHallam) [inline config/routes.rb:366]: suspect this comment should be removed

File: db/migrate/20220825093553_split_world_location_news_page.rb
Comment (JonathanHallam) [inline db/migrate/20220825093553_split_world_location_news_page.rb:38]: minor - maybe add a line in the commit message on why this change is irreversible

File: test/functional/admin/world_location_news_controller_test.rb
Comment (JonathanHallam) [inline test/functional/admin/world_location_news_controller_test.rb:59]: 😬

File: test/integration/world_location_integration_test.rb
Comment (JonathanHallam) [inline test/integration/world_location_integration_test.rb:33]: 🇫🇷

File: config/routes.rb
Comment (brucebolt) [reply config/routes.rb:366]: Ah, yes, it should've been.  I'll remove it now.

File: db/migrate/20220825093553_split_world_location_news_page.rb
Comment (brucebolt) [reply db/migrate/20220825093553_split_world_location_news_page.rb:38]: I've split the migration out into a separate commit and added an explanation as to why it's irreversible.  It probably would be possible to make it reversible, but the migration would become very complex (since we'd need to somehow recombine the translations from two different tables).

File: test/support/govspeak_validation_test_helper.rb
Comment (JonathanHallam) [inline test/support/govspeak_validation_test_helper.rb]: why have changes been made to this file?

Comment (brucebolt) [reply test/support/govspeak_validation_test_helper.rb]: The method previously made an assumption that you wanted to run the XSS tests against the factory whose name matched the class being tested (e.g. the `:world_location_news` factory would be used for the `WorldLocationNews` model).  However we want to test against the `:world_location_news_with_stubbed_slug` factory instead, so can't use this assumption anymore.

We shifted this to always be the first argument to the method as it's no longer optional and would've had to be utilised by appending `supplied_factory_name: :foo` to the method calls, because of the usage of the splat operator prior to this.  This change means we can just add it as the first argument.

File: config/locales/lt.yml
Comment (JonathanHallam) [inline config/locales/lt.yml:48]: extra whitespace

File: config/locales/pa-pk.yml
Comment (JonathanHallam) [inline config/locales/pa-pk.yml:48]: whitespace

File: config/locales/ja.yml
Comment (JonathanHallam) [inline config/locales/ja.yml:48]: whitespace

**Review by JonathanHallam — APPROVED** [review]
Approving this, I'm adding a do not merge flag to this so that we can do some testing on integration immediately before merging

File: config/locales/ja.yml
Comment (brucebolt) [reply config/locales/ja.yml:48]: Looks like this was added in when I normalized the locale files.  I'll remove the whitespace here and add the normalization as either another PR or another commit.

File: config/locales/pa-pk.yml
Comment (brucebolt) [reply config/locales/pa-pk.yml:48]: 👍

**Pull request merged**
